Source/automap.cpp

```diff
@@ -707,15 +724,21 @@ void DrawAutomap(const Surface &out)
 	DrawAutomapText(out);
 }
 
-void SetAutomapView(Point position)
+void UpdateAutomapExplorer(Point map, MapExplorationType explorer)
+{
+	if (AutomapView[map.x][map.y] < explorer)
+		AutomapView[map.x][map.y] = explorer;

```

Comment:
I think we should be more explicit about this "hierarchy" of values you are implicitly creating here with the comparisons.

This feels a bit too magical to me.

Reply:
why? higher "explorer" takes priority = 
none < shrine < other players in multi < self
to provide proper map coloring when you explore map with a different color already